src/python/pants_release/release.py

```diff
@@ -536,44 +505,50 @@ def build_pants_wheels() -> None:
         *(package.target for package in PACKAGES),
     )
 
-    with set_pants_version(CONSTANTS.pants_unstable_version):
-        try:
-            subprocess.run(args, check=True)
-        except subprocess.CalledProcessError as e:
-            failed_packages = ",".join(package.name for package in PACKAGES)
-            failed_targets = " ".join(package.target for package in PACKAGES)
+    try:
+        subprocess.run(args, check=True)
+    except subprocess.CalledProcessError as e:
+        failed_packages = ",".join(package.name for package in PACKAGES)
+        failed_targets = " ".join(package.target for package in PACKAGES)
+        die(
+            softwrap(
+                f"""
+                Failed to build packages {failed_packages} for {version} with targets
+                {failed_targets}.
+
+                {e!r}
+                """
+            )
+        )
+
+    # TODO(#10718): Allow for sdist releases. We can build an sdist for
```

Comment:
While you're here, if you happen to make other changes, let's delete this TODO? https://github.com/pantsbuild/pants/issues/10718 was closed as not relevant to the new distribution model. Not worth a new CI round, so only if you happen to be making changes before merge anyway.